### PR TITLE
feat: LT-1299 - countersigning queue

### PR DESCRIPTION
**Pull request by Mark-Spiers-kainos**

### CODE REVIEW

File: ui_automation_tests/pages/queues_pages.py [hunk 1]
```diff
@@ -8,6 +9,7 @@ class QueuesPages(BasePage):
     QUEUE_CHECKBOXES = ".govuk-checkboxes .govuk-checkboxes__input"  # CSS
     TABLE_ROWS = ".govuk-table__body .govuk-table__row"  # CSS
     LINK_EDIT_SELECTOR = '.govuk-link[href*="edit"]'
+    SELECT_COUNTERSIGNING_QUEUE_ID = "countersigning_queue"
```
Comment (jfaracik) [inline ui_automation_tests/pages/queues_pages.py:12]: IDs use dashes, not underscores :)

File: queues/services.py [hunk 2]
```diff
@@ -7,8 +7,11 @@
 from http import HTTPStatus
 
 
-def get_queues(request, disable_pagination=True, page=1, convert_to_options=False):
-    data = get(request, QUEUES_URL + f"?page={page}&disable_pagination={disable_pagination}").json()
+def get_queues(request, disable_pagination=True, page=1, convert_to_options=False, users_team_first=False):
+    data = get(
+        request,
+        QUEUES_URL + f"?page={page}&disable_pagination={disable_pagination}&users_team_first={users_team_first}",
+    ).json()
```
Comment (jfaracik) [inline queues/services.py:14]: Using convert_parameters_to_query_params and locals would be cleaner :)

```
data = get(request, QUEUES_URL + convert_parameters_to_query_params(locals()).json()
```

File: queues/forms.py [hunk 3]
```diff
@@ -1,22 +1,56 @@
...
     return Form(
         title=AddQueueForm.TITLE,
         description=AddQueueForm.DESCRIPTION,
-        questions=[TextInput(title=AddQueueForm.Name.TITLE, description=AddQueueForm.Name.DESCRIPTION, name="name",)],
+        questions=[
+            TextInput(title=AddQueueForm.Name.TITLE, description=AddQueueForm.Name.DESCRIPTION, name="name",),
+            Select(
+                title=AddQueueForm.CountersigningQueue.TITLE,
+                description=AddQueueForm.CountersigningQueue.DESCRIPTION,
+                options=get_queues(
```
Comment (BenAndrew-Kainos) [inline queues/forms.py:17]: This also needs `remove_current_queue_id`

File: ui_automation_tests/features/queues.feature [hunk 4]
```diff
@@ -51,3 +51,22 @@ Feature: I want to define new work queues and the teams they belong to
...
+
+  @LT_1299_countersigning_queues_working @regression
+  Scenario: Finish with a team queue and have countersigning queue automatically apply
+    Given I sign in to SSO or am signed into SSO
+    And I create open application or open application has been previously created
+    And a queue has been created
+    And a new countersigning queue has been created
+    When I go to queues
+    Then I see my queue
+    When I edit the new queue with a countersigning queue
```
Comment (BenAndrew-Kainos) [inline ui_automation_tests/features/queues.feature:63]: Also check that you can see the countersigning queue in the queues list

File: queues/forms.py (hunk 3 above)
Comment (Mark-Spiers-kainos) [reply queues/forms.py:17]: This is for creating, so there is no current queue id

File: ui_automation_tests/features/queues.feature (hunk 4 above)
Comment (Mark-Spiers-kainos) [reply ui_automation_tests/features/queues.feature:64]: added step before `Then I see my queue` as to check countersigning queue exists.

File: queues/services.py (hunk 2 above)
Comment (Mark-Spiers-kainos) [reply queues/services.py]: done

File: ui_automation_tests/pages/queues_pages.py (hunk 1 above)
Comment (Mark-Spiers-kainos) [reply ui_automation_tests/pages/queues_pages.py:12]: As discussed the Select fields uses the name to set as id, so can't easily change this.

File: ui_automation_tests/features/queues.feature (hunk 4 above)
Comment (BenAndrew-Kainos) [reply ui_automation_tests/features/queues.feature:64]: I meant that when you add the countersigning queue to the queue, you see that it has a countersigning queue

**Pull request merged**